destination/batches.go

```diff
@@ -119,3 +121,46 @@ func parseBatches(records []sdk.Record, parser streamNameParser) ([]recordBatch,
 
 	return batches, nil
 }
+
+type streamProvisioner struct {
+	existingStreams cmap.ConcurrentMap[string, bool]
+}
+
+func newStreamProvisioner() *streamProvisioner {
+	return &streamProvisioner{existingStreams: cmap.New[bool]()}
+}
+
+func (s *streamProvisioner) ensureStreamsExist(
+	ctx context.Context,
+	destination *Destination,
+	batches []recordBatch,
+) error {
+	for _, batch := range batches {
+		streamName := batch.streamName
+		if _, ok := s.existingStreams.Get(streamName); ok {
+			continue
+		}
+
+		ok, err := destination.doesStreamExist(ctx, streamName)
```

Comment:
I think we should check for the auto configuration option here to then create the stream if it doesn't exist, no?

Reply:
It is handled in an upper layer, here:

https://github.com/conduitio-labs/conduit-connector-kinesis/blob/9c90270869e0513c631d5b3901946dbef57a43a2/destination/destination.go#L275-L280

Reply:
ensureStreamsExist would have strange naming otherwise, allowing streams to not exist.